src/bun.js/api/bun/subprocess.zig

```diff
@@ -1302,13 +1304,11 @@ pub const Subprocess = struct {
             return .zero;
         };
 
-        if (env_array.items.len > 0) {
-            env_array.append(allocator, null) catch {
-                globalThis.throw("out of memory", .{});
-                return .zero;
-            };
-            env = @as(@TypeOf(env), @ptrCast(env_array.items.ptr));
-        }
+        env_array.append(allocator, null) catch {
```

Comment:
https://github.com/oven-sh/bun/blob/36f9d2291c71d151dceff6d7d40adbbb2c3b3809/src/bun.js/api/bun/spawn.zig#L192-L197

`null` should be appended, even if it's empty.